lib/user.rb

```diff
@@ -0,0 +1,31 @@
+require 'minitest/autorun'
+require 'minitest/pride'
+require './lib/joke'
+require 'pry'
+
+class User
+  attr_reader :name, :jokes
+
+  def initialize(name)
+    @name = name
+    @jokes = []
+  end
+
+  def learn(joke)
+    @joke = joke
+    @jokes << @joke
+  end
+
+  def sal_tells(name, joke)
+    @name = name
+    @joke = joke
+    @jokes << @joke
+  end
+
+  def joke_by_id(id)
+    if @id == id
+      @joke_1
+    end
```

Comment:
This is close. We're in our User object though, which doesn't have an ID value. We'd have to search through this User's jokes to see if any jokes match the ID value we're passing in.